compiler/src/dotty/tools/dotc/inlines/Inliner.scala

```diff
@@ -399,7 +399,9 @@ class Inliner(val call: tpd.Tree)(using Context):
    *  type aliases, add proxy definitions to `opaqueProxies` that expose these aliases.
    */
   private def addOpaqueProxies(tp: Type, span: Span, forThisProxy: Boolean)(using Context): Unit =
-    tp.foreachPart {
+    val foreachTpPart =
+      (p: Type => Unit) => if (forThisProxy) tp.foreachPartWithoutTypeParams(p) else tp.foreachPart(p)
```

Comment:
I wouldn't introduce `foreachPartWithoutTypeParams` for one use case.  Also, it's a confusing name, because you're skipping applied type arguments, rather than "type parameters".